kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/EncryptionSchemeTest.java

```diff
@@ -19,19 +22,49 @@ class EncryptionSchemeTest {
     void shouldRejectInvalidConstructorArgs() {
         EnumSet<RecordField> nonEmpty = EnumSet.of(RecordField.RECORD_VALUE);
         var empty = EnumSet.noneOf(RecordField.class);
-        assertThrows(NullPointerException.class, () -> new EncryptionScheme<>(null, nonEmpty));
+        assertThrows(NullPointerException.class, () -> new EncryptionScheme(null, nonEmpty));
         Object kekId = new Object();
-        assertThrows(NullPointerException.class, () -> new EncryptionScheme<>(kekId, null));
-        assertThrows(IllegalArgumentException.class, () -> new EncryptionScheme<>(kekId, empty));
+
+        final MyKekId myKekId = new MyKekId(kekId);
+
+        assertThrows(NullPointerException.class, () -> new EncryptionScheme(myKekId, null));
+        assertThrows(IllegalArgumentException.class, () -> new EncryptionScheme(myKekId, empty));
     }
 
     @Test
     void shouldAcceptValidConstructorArgs() {
         EnumSet<RecordField> nonEmpty = EnumSet.of(RecordField.RECORD_VALUE);
         Object kekId = new Object();
-        var es = new EncryptionScheme<>(kekId, nonEmpty);
-        assertEquals(kekId, es.kekId());
+        var es = new EncryptionScheme(new MyKekId(kekId), nonEmpty);
+        assertEquals(new MyKekId(kekId), es.kekId());
         assertEquals(nonEmpty, es.recordFields());
     }
 
-}
+    private record MyKekId(Object kekId) implements KekId {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <K> K getId(Class<K> keyType) {
+            return (K) kekId;
+        }
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
What's stopping this record relying on the record's default equality?

Reply:
That's a good point it's a hanger over from being a class.

Reply:
resolved with 5a11ea4e88cf4219528390d5b32788a5a1c34807